pkg/resources/kafka/configmap.go

```diff
@@ -346,6 +346,44 @@ func generateListenerSSLConfig(config *properties.Properties, name string, sslCl
 	}
 }
 
+// mergeSuperUsersPropertyValue merges the super.users property value from the source into the target properties
+func mergeSuperUsersPropertyValue(source *properties.Properties, target *properties.Properties) {
```

Comment:
**Recommended**: In Go it is more idiomatic to pass two values and return a third value which can be used to overwrite any of the inputs at the caller side if needed (or assign to a new variable) instead of creating side-effect ridden functions with pseudo-references (pointers).
For example append works similarly `newSlice = append(oldSlice, newValue)`.

Is there a reason to prefer writing pointers as a side effect in a method here compared to using a proper function and returning a new value?

Reply:
I wanted to put everything in that mergeSuperUsersPropertyValue function thus it works almost the same as the finalBrokerConfig.Merge(opGenConf).

It can be a good idea to follow the append convention. Like  mergeSuperUsersPropertyValue(source *properties.Properties, superusers) return *properties.Properties